openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -804,7 +805,11 @@ def setUp(self):
 
     def _assert_supplemented(self, actual, **kwargs):
         """DRY helper used to verify that program data is extended correctly."""
-        self.course_run.update(
+        program = deepcopy(self.program)
+        course_run = deepcopy(self.course_run)
+        course = deepcopy(self.catalog_course)
```

Comment:
Why do we need to deepcopy them?

Reply:
@ahsan-ul-haq Actually when we update the `self.course_run` or `self.program` the changes were reflecting to actual data which actually we are asserting, due to that this test was always passing and never gonna fail that's why I used deepcopy to create copy of the program data and then update it according to the expected data returned by ProgramDataExtender.